File: platform/android/lib/src/java/com/github/aui/android/AuiActivity.java
package com.github.aui.android;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.os.Build;
import android.os.Bundle;
import android.view.WindowManager;

public class AuiActivity extends AppCompatActivity {
    private String getApplicationName() {
        Context ctx = getApplicationContext();
        ApplicationInfo applicationInfo = ctx.getApplicationInfo();
        int id = applicationInfo.labelRes;
        return id == 0 ? applicationInfo.nonLocalizedLabel.toString() : ctx.getString(id);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        getWindow().setFlags(WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS,
                WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS);

        AUI.ourContext = this;

        MyGLSurfaceView surfaceView = new MyGLSurfaceView(this);
        setContentView(surfaceView);
    }
}

File: platform/android/src/main/java/com/github/aui/android/AUIActivity.kt
/*
 * AUI Framework - Declarative UI toolkit for modern C++20
 * Copyright (C) 2020-2025 Alex2772 and Contributors, Magicsea
 *
 * SPDX-License-Identifier: MPL-2.0
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package com.github.aui.android

import android.os.Bundle
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import kotlin.system.exitProcess

open class AUIActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.setFlags(
            WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS,
            WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS
        )
        Platform.ourContext = this
        val surfaceView = AUIView(this)
        setContentView(surfaceView)
        System.loadLibrary("app")
    }

    override fun onDestroy() {
        super.onDestroy()

        exitProcess(0) // force the process to finish properly
    }
}